src/Types/BlockItemSummary.cs

```diff
@@ -0,0 +1,368 @@
+    /// <exception cref="MissingEnumException{DetailsOneofCase}">Throws exception when returned type not known</exception>
+    public bool IsSuccess() =>
+        this._blockItemSummary.DetailsCase switch
+        {
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.AccountTransaction =>
+                !AccountTransactionDetailsIsRejected(this._blockItemSummary.AccountTransaction, out _),
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.AccountCreation => true,
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.Update => true,
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.None =>
+                throw new MissingEnumException<Grpc.V2.BlockItemSummary.DetailsOneofCase>(Grpc.V2.BlockItemSummary.DetailsOneofCase.None),
+            _ =>
+                throw new MissingEnumException<Grpc.V2.BlockItemSummary.DetailsOneofCase>(this._blockItemSummary.DetailsCase)
+        };
+
+    /// <summary>
+    /// Return whether the transaction has failed to achieve the intended
+    /// effects.
+    /// </summary>
+    /// <returns>bool representing if transaction is rejected</returns>
+    public bool IsReject() => this.TryGetRejectedAccountTransaction(out _);
+
+    /// <summary>
+    /// When transaction is rejected this method will return true and rejected reason
+    /// will be return in out param.
+    /// </summary>
+    /// <param name="rejectReason">Output with rejected reason if rejected.</param>
+    /// <returns>bool representing if transaction is rejected.</returns>
+    /// <exception cref="MissingEnumException{DetailsOneofCase}">Throws exception when returned type not known</exception>
+    public bool TryGetRejectedAccountTransaction(out RejectReason? rejectReason)
+    {
+        rejectReason = null;
+        return this._blockItemSummary.DetailsCase switch
+        {
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.AccountTransaction =>
+                AccountTransactionDetailsIsRejected(this._blockItemSummary.AccountTransaction, out rejectReason),
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.AccountCreation => false,
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.Update => false,
+            Grpc.V2.BlockItemSummary.DetailsOneofCase.None =>
+                throw new MissingEnumException<Grpc.V2.BlockItemSummary.DetailsOneofCase>(Grpc.V2.BlockItemSummary.DetailsOneofCase.None),
+            _ =>
+                throw new MissingEnumException<Grpc.V2.BlockItemSummary.DetailsOneofCase>(this._blockItemSummary.DetailsCase)
+        };
+    }
+
+    /// <summary>
+    /// Returns true and set sender account if of account transaction type.
+    /// </summary>
+    /// <param name="sender">Output given when sender present.</param>
+    /// <returns>If sender account present.</returns>
+    /// <exception cref="MissingEnumException{DetailsOneofCase}">Throws exception when returned type not known</exception>
+    public bool TryGetSenderAccount(out AccountAddress? sender)
```

Comment:
Same here. Is it not more ergonomic to return values seeing that the caller does not have to create the variable first?

Reply:
AccountAddress is nullable and hence the caller would not need to create a variable.

Reply:
I don't understand. To get anything out they need to do something like
```
var address;
if TryGetSenderAccount(out address) {
    ...
}
```

I had assumed there would at least be a standard Optional<T> type that would make this cleaner. BUt if there isn't then I suppose we must live with this C-style stuff.

Reply:
I believe this is the way to do it - somehow one needs to make a check. I can give a review with Ali and see if he knows a better pattern.

To me it seems  however quite much the same between C# and Rust

C#

```
if (TryGetSenderAccount(out var address) {
   // do something with address
}
```

Rust 

```
if let Some(address) = senderAccount() {
   // do something with address
}
```

please let me know if I’m missing something - it always fun fun to learn :-)

Reply:
I did not know you can declare variables inside parameters to a function call.

Now that I do know I hate it since it adds complexity. :P 

But OK.

Also note in Rust you would probably do
```

```
let Some(address) = senderAccount() else {
    // Handle failure.
};
// use address here
```
to avoid too much nesting.

Reply:
I didn’t know you could do the else afterwards - nice since nesting is exactly one of the things in Rust I find difficult and make it difficult to read :-)